cpu/atmega_common/periph/gpio.c

```diff
@@ -133,9 +133,9 @@ int gpio_init(gpio_t pin, gpio_mode_t mode)
             break;
         case GPIO_IN:
             _SFR_MEM8(_ddr_addr(pin)) &= ~(1 << _pin_num(pin));
-            _SFR_MEM8(_port_addr(pin)) &= ~(1 << _pin_num(pin));
```

Comment:
Controlling just the data direction without controlling the port is a way to emulate an open- drain-output on atmega and since no comment about GPIO_IN states that it has no pull-up i think it would be a good idea to remove the line and keep the change fast, since no one asked to touch the pull-up configuration. 

But the variant of GPIO_IN with setting port to 0 and hence removing the pull-up in case is also compatible to the open-drain usage (only some extra processor cycles) and i don't even know what the invers (pull hard to VCC or let the port float to ground), which is harmed by this line, would be called, i would also be OK with reintroducing the line, if you request it after reading this comment.

I would prefer the faster variant without changing the port.

Reply:
Stated in ```periph/gpio.h```: ```GPIO_IN ,  /**< configure as input without pull resistor */```

```GPIO_OD``` and ```GPIO_OD_PU``` are currently unsupported by the ATmegas. Adding support would involve toggling direction as you say, but would also require a bit of RAM to hold the extra mode bits to implement properly. Doing strange calls to ```gpio_init``` is not the way to do it.

Also, I don't think speed is a priority with ```gpio_init```, because it is not expected to be called that often.

Reply:
you are right, i overlooked it
i changed the behavior back and introduced an optimization calculate the pin mask just once 

avr-nm before: 
00000130 000000b6 T gpio_init
and after:
00000130 0000008a T gpio_init